### ISSUE TITLE
angle brackets are not being matched properly, causing keywords to be highlighted as a type

### ISSUE THREAD

**Issue by DibyodyutiMondal** [typescript-language-server/typescript-language-server#753]
I am not sure if this is a problem with vs code or the language service, but I'm assuming it is the language service. Apologies in advance if I'm wrong.

The following snippet works fine, and the code highlighting in the editor works fine too, except when I wrap the type parameter.

### snippet
```typescript
export const mapResponse = {
  blogPost: <I extends { created: string; updated: string; }>(input: I) => {
    return {
      ...input,
      created: DateTime.fromISO(input.created),
      updated: DateTime.fromISO(input.updated)
    };
  }
};
```
#### `return` keyword is correctly identified as a keyword
![image](https://github.com/typescript-language-server/typescript-language-server/assets/50345677/92707eff-52c4-4a02-b4c1-97a5a91e1e70)

#### `return` keyword is incorrectly identified as a type(?)
![image](https://github.com/typescript-language-server/typescript-language-server/assets/50345677/9965189b-6936-4b67-a21e-eaa9b616a7cb)

**Comment by rchl** [comment]
It's not language server applying syntax highlighting, it's the editor.

Language server can augment the colors through semantic highlighting but it doesn't apply semantic tokens to keywords like `return` so it's not gonna help in this case.

You should report this in the VSCode repo. VSCode doesn't even use this server.
